pyvista/plotting/render_passes.py

```diff
@@ -0,0 +1,255 @@
+        """Add a vtkGaussianBlurPass pass."""
+        if self._blur_passes:
+            self._remove_pass(self._blur_passes.pop(0))
+
+    def enable_shadow_pass(self):
+        """Enable shadow pass."""
+        # shadow pass can be directly added to the base pass collection
+        if self._shadow_map_pass is not None:
+            return
+        self._shadow_map_pass = _vtk.vtkShadowMapPass()
+        self._pass_collection.AddItem(self._shadow_map_pass.GetShadowMapBakerPass())
+        self._pass_collection.AddItem(self._shadow_map_pass)
+        self._update_passes()
+        return self._shadow_map_pass
+
+    def disable_shadow_pass(self):
+        """Disable shadow pass."""
+        if self._shadow_map_pass is None:
+            return
+        self._pass_collection.RemoveItem(self._shadow_map_pass.GetShadowMapBakerPass())
+        self._pass_collection.RemoveItem(self._shadow_map_pass)
+        self._update_passes()
+
+    def enable_depth_of_field_pass(self, automatic_focal_distance=True):
+        """Enable the depth of field pass."""
+        if self._dof_pass is not None:
+            return
+        self._dof_pass = _vtk.vtkDepthOfFieldPass()
+        self._dof_pass.SetAutomaticFocalDistance(automatic_focal_distance)
+        self._add_pass(self._dof_pass)
+        return self._dof_pass
+
+    def disable_depth_of_field_pass(self):
+        """Disable the depth of field pass."""
+        if self._dof_pass is None:
+            return
+        self._remove_pass(self._dof_pass)
+        self._dof_pass = None
+
+    def enable_ssaa_pass(self):
+        """Enable super space anti-aliasing pass."""
+        if self._ssaa_pass is not None:
+            return
+        self._ssaa_pass = _vtk.vtkSSAAPass()
+        self._add_pass(self._ssaa_pass)
+        return self._ssaa_pass
+
+    def disable_ssaa_pass(self):
+        """Disable super space anti-aliasing pass."""
+        if self._ssaa_pass is None:
+            return
+        self._remove_pass(self._ssaa_pass)
+        self._ssaa_pass = None
+
+    def _update_passes(self):
+        """Reassemble pass delegation."""
+        if self._renderer is None:
+            raise RuntimeError('The renderer has been closed.')
+
+        current_pass = self._camera_pass
+        for class_name in PRE_PASS + POST_PASS:
+            if class_name in self._passes:
+                for render_pass in self._passes[class_name]:
+                    render_pass.SetDelegatePass(current_pass)
+                    current_pass = render_pass
+
+        # reset to the default rendering if no special passes have been added
+        if isinstance(current_pass, _vtk.vtkCameraPass) and self._shadow_map_pass is None:
+            self._renderer.SetPass(None)
+        else:
+            self._renderer.SetPass(current_pass)
+
+    @staticmethod
+    def _class_name_from_vtk_obj(obj):
+        """Return the class name from a vtk object."""
+        return str(type(obj)).split('.')[-1].split("'")[0]
```

Comment:
I think this should work:
```suggestion
        return type(obj).__name__
```
Of course if this works we might not need a staticmethod for it.

Reply:
There is also the `GetClassName()` all all `vtkObject`s

Reply:
Went with bane's approach in 69243a9ff453f7d292ce9329026b0ef8c43fa355.